libs/config/value.go

```diff
@@ -175,3 +184,51 @@ func (v Value) MustFloat() float64 {
 func (v Value) MustTime() time.Time {
 	return v.v.(time.Time)
 }
+
+func (v Value) MarshalYAML() (interface{}, error) {
+	if v.Kind() == KindMap {
+		m, _ := v.AsMap()
+		keys := maps.Keys(m)
+		// We're using location lines to define the order of keys in YAML.
+		// The location is set when we convert API response struct to config.Value representation
+		// See convert.convertMap for details
+		sort.SliceStable(keys, func(i, j int) bool {
+			return m[keys[i]].Location().Line < m[keys[j]].Location().Line
+		})
+
+		content := make([]*yaml.Node, 0)
+		for _, k := range keys {
+			item := m[k]
+			node := yaml.Node{Kind: yaml.ScalarNode, Value: k}
+			in, err := item.MarshalYAML()
+			c := in.(*yaml.Node)
+			if err != nil {
+				return nil, err
+			}
+			content = append(content, &node)
+			content = append(content, c)
+		}
+
+		return &yaml.Node{Kind: yaml.MappingNode, Content: content}, nil
+	}
+
+	if v.Kind() == KindSequence {
+		s, _ := v.AsSequence()
+		content := make([]*yaml.Node, 0)
+		for _, item := range s {
+			in, err := item.MarshalYAML()
+			c := in.(*yaml.Node)
+			if err != nil {
+				return nil, err
+			}
+			content = append(content, c)
+		}
+		return &yaml.Node{Kind: yaml.SequenceNode, Content: content}, nil
+	}
+
+	return &yaml.Node{Kind: yaml.ScalarNode, Value: fmt.Sprint(v.AsAny())}, nil
```

Comment:
Is everything else really a scalar? Would be good to validate this with a unit test.

To convey the same here in the code, you could use a switch/case on the kind of value.